compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/WebImeInputService.kt

```diff
@@ -0,0 +1,172 @@
+        val htmlInput = document.createElement("textarea") as HTMLTextAreaElement
+
+        htmlInput.setAttribute("autocorrect", "off")
+        htmlInput.setAttribute("autocomplete", "off")
+        htmlInput.setAttribute("autocapitalize", "off")
+        htmlInput.setAttribute("spellcheck", "false")
+
+        val inputMode = when (imeOptions.keyboardType) {
+            KeyboardType.Text -> "text"
+            KeyboardType.Ascii -> "text"
+            KeyboardType.Number -> "number"
+            KeyboardType.Phone -> "tel"
+            KeyboardType.Uri -> "url"
+            KeyboardType.Email -> "email"
+            KeyboardType.Password -> "password"
+            KeyboardType.NumberPassword -> "number"
+            KeyboardType.Decimal -> "decimal"
+            else -> "text"
+        }
+
+        val enterKeyHint = when (imeOptions.imeAction) {
+            ImeAction.Default -> "enter"
+            ImeAction.None -> "enter"
+            ImeAction.Done -> "done"
+            ImeAction.Go -> "go"
+            ImeAction.Next -> "next"
+            ImeAction.Previous -> "previous"
+            ImeAction.Search -> "search"
+            ImeAction.Send -> "send"
+            else -> "enter"
+        }
+
+        htmlInput.setAttribute("inputmode", inputMode)
+        htmlInput.setAttribute("enterkeyhint", enterKeyHint)
+
+        htmlInput.style.apply {
+            setProperty("position", "absolute")
+            setProperty("user-select", "none")
+            setProperty("forced-color-adjust", "none")
+            setProperty("white-space", "pre-wrap")
+            setProperty("align-content", "center")
+            setProperty("top", "0")
+            setProperty("left", "0")
+            setProperty("padding", "0")
+            setProperty("opacity", "0")
+            setProperty("color", "transparent")
+            setProperty("background", "transparent")
+            setProperty("caret-color", "transparent")
+            setProperty("outline", "none")
+            setProperty("border", "none")
+            setProperty("resize", "none")
+            setProperty("text-shadow", "none")
+        }
+
+        htmlInput.addEventListener("input", {
+                htmlInput
+                val text = htmlInput.value
+                val cursorPosition = htmlInput.selectionEnd
+                sendImeValueToCompose(onEditCommand, text, cursorPosition)
+        })
+
+        return htmlInput
+    }
+
+    private fun sendImeValueToCompose(
+        onEditCommand: (List<EditCommand>) -> Unit,
+        text: String,
+        newCursorPosition: Int? = null
+    ) {
+        val value = if (text == "\n") {
+            ""
+        } else {
+            text
+        }
+
+        if (newCursorPosition != null) {
+            onEditCommand(
+                listOf(
+                    DeleteAllCommand(),
+                    CommitTextCommand(value, 1),
+                    SetSelectionCommand(newCursorPosition, newCursorPosition)
```

Comment:
Would it work if instead of adding 'SetSelectionCommand', it would pass newCursorPosition to CommitTextCommand (2nd parameter)?

```
class CommitTextCommand(
    val annotatedString: AnnotatedString,
    val newCursorPosition: Int // here?
) 
```